code/game/dna/mutations/monkey_mutation.dm

```diff
@@ -26,7 +26,22 @@
 	H.invisibility = 101
 	var/has_primitive_form = H.dna.species.primitive_form // cache this
 	if(has_primitive_form)
-		H.set_species(has_primitive_form, keep_missing_bodyparts = TRUE)
+
+		if(ispath(has_primitive_form, /mob/living/simple_animal/nian_caterpillar))
+			var/datum/mind/M = H.mind
+			//addtimer(CALLBACK(src, PROC_REF(become_headslug), user), 5 SECONDS)
+			var/mob/living/simple_animal/nian_caterpillar/caterpillar = new(get_turf(H))
+			//var/obj/item/organ/internal/eyes/E = H.get_int_organ(/obj/item/organ/internal/eyes)
+			new /obj/effect/temp_visual/monkeyify(H.loc)
+			caterpillar.origin = M
+			caterpillar.origin.transfer_to(caterpillar)
+			caterpillar.real_name = H.name
+			caterpillar.name = caterpillar.real_name
+			//worme.color = E.eye_color
```

Comment:
and here

Reply:
Old code, caterpillar's looked scuffed with their color being changed